lib/mayaUsd/resources/scripts/mayaUsdAddMayaReference.py

```diff
@@ -49,47 +53,158 @@ def createPrimAndAttributes(stage, primPath, mayaReferencePath, mayaNamespace, m
+    # If a prim with this name already exists, then add a numeric suffix after "RN".
+    startName = mayaNamespace + "RNgroup"
+    checkName = mayaUsd.ufe.uniqueChildName(parentPrim, startName)
+    index = 1
+    while (checkName != startName):
+        startName = "%sRN%dgroup" % (mayaNamespace, index)
+        checkName = mayaUsd.ufe.uniqueChildName(parentPrim, startName)
+        index += 1
+    return startName
+
 # Adapted from testMayaUsdSchemasMayaReference.py.
 def createMayaReferencePrim(ufePathStr, mayaReferencePath, mayaNamespace, 
                             mayaReferencePrimName = kDefaultMayaReferencePrimName,
-                            variantSetName = kDefaultVariantSetName,
-                            variantName = kDefaultVariantName,
+                            groupPrim = None, variantSet = None,
                             mayaAutoEdit = kDefaultEditAsMayaData):
-    '''Create a Maya reference prim parented to the argument path.'''
+    '''Create a Maya reference prim and optional group prim parented to the argument path.
+    Optionally create a variant set and name and placed the edits inside that variant.
+
+    Naming of Maya Reference prim is supported, otherwise default name is used.
+
+    The group prim is optional.
+
+    The variant set and name are optional
+
+    Parameters:
+    -----------
+    ufePathStr : str : Ufe PathString of parent prim to add Maya Reference
+    mayaReferencePath : str : File path of Maya Reference (for attribute)
+    mayaNamespace : str : Namespace (for attribute)
+    mayaReferencePrimName : str [optional] : Name for the Maya Reference prim
+    groupPrim : tuple(str,str,str) [optional] : The Group prim Name, Type & Kind to create
+                                                Note: the name is optional and will be auto-computed
+                                                      if empty or not provided.
+                                                Note: Type and Kind are both mandatory, but Kind is
+                                                      allowed to be empty string.
+    variantSet : tuple(str,str) [optional] : The Variant Set Name and Variant Name to create
+
+    Return:
+    -------
+    The Usd prim of the newly created Maya Reference or an invalid prim if there is an error.
+    '''
 
     # Make sure the prim name is valid and doesn't already exist.
     parentPrim = mayaUsd.ufe.ufePathToPrim(ufePathStr)
 
-    checkName = mayaUsd.ufe.uniqueChildName(parentPrim, mayaReferencePrimName)
+    # There are special conditions when we are given the ProxyShape gateway node.
+    ufePath = ufe.PathString.path(ufePathStr)
+    isGateway = (ufePath.nbSegments() == 1)
+
+    # Were we given a Group prim to create?
+    groupPrimName = None
+    groupPrimType = None
+    groupPrimKind = None
+    if groupPrim:
+        if (len(groupPrim) == 2):
+            groupPrimType, groupPrimKind = groupPrim
```

Comment:
The group prim name is optional - if not given one will be auto-generated. See `getDefaultGroupPrimName()` above.